src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/MetricsService.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.finrem.caseorchestration.config.CourtDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.config.CourtDetailsConfiguration;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseData;
+
+import java.util.Optional;
+
+import static java.util.Optional.ofNullable;
+
+@Service
+@RequiredArgsConstructor
+public class MetricsService {
+
+    private final CourtDetailsConfiguration courtDetailsConfiguration;
+
+    /**
+    * Sets PowerBI fields used for tracking and gather statistics.
+    * Though the fields are prefixed Consent_Order_FRC, these tracking fields are used for both consented and
+    * contested applications.
+    *
+    * @param caseData instance of FinremCaseData
+    */
+    public void  setCourtMetrics(FinremCaseData caseData) {
+
+        String selectedCourtId = caseData.getSelectedAllocatedCourt();
+
+        Optional<CourtDetails> selectedCourtDetails = ofNullable(courtDetailsConfiguration.getCourts().get(selectedCourtId));
+
+        String getEmailForSelectedCourt = selectedCourtDetails.map(CourtDetails::getEmail).orElse("");
+        String getNameForSelectedCourt = selectedCourtDetails.map(CourtDetails::getCourtName).orElse("");
+        String getAddressForSelectedCourt = selectedCourtDetails.map(CourtDetails::getCourtAddress).orElse("");
+        String getPhoneForSelectedCourt = selectedCourtDetails.map(CourtDetails::getPhoneNumber).orElse("");
+
```

Comment:
I am curious if empty strings would cause problem.

Reply:
I'd be temped to throw an exception. Why would you not get a court?

Reply:
Hi @so99y.  Hope this answers your question.  CCD can store the empty strings - just tested, this is what I get locally:

  "consentOrderFRCEmail": "",
  "consentOrderFRCPhone": "",
  "consentOrderFRCName": "",
  "consentOrderFRCEmail": "",

Does that answer your question?

Reply:
@so99y @jthmcts So this is grabbing court information to forward for Power Bi metrics.

Would we want to stop a user for that?